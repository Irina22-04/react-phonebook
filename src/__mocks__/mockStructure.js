export const mockStructure = [
    {
        "name": "OOO 'Дети дикой обезьяны'",
        "id": 1
    },
    {
        "name": "Управление 1",
        "parent": 1,
        "id": 2
    },
    {
        "name": "Управление 22",
        "parent": 1,
        "id": 3
    },
    {
        "name": "Управление 32",
        "parent": 1,
        "id": 4
    },
    {
        "name": "Отдел по работе с клиентами",
        "id": 5,
        "parent": 2
    },
    {
        "name": "Отдел по работе с юридическими лицами",
        "id": 8,
        "parent": 2
    }
];