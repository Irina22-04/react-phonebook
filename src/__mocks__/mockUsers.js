export const mockUsers = [
    {
        "name": "Дмитрий",
        "surname": "Дмитриевич",
        "patronym": "Дмитриевич",
        "department": 8,
        "fullname": "Дмитриев Дмитрий Дмитриевич",
        "id": 104,
        "isAdmin": true,
        "email": "admin@mail.ru",
        "password": "$2a$10$AnPxLMUGPKB8eAsdF90pFOWi/h2ejtPhrkzA6.a/nNUty8SexQBQe",
        "position": "специалист",
        "phone": "340-34-34",
        "businessMobile": "(33)333-33-33",
        "personalMobile": "(44)444-44-44",
        "personalEmail": "Dmit@mail.ru",
        "birthday": "22/02/1999"
    },
    {
        "name": "Михаил",
        "surname": "Михайлов",
        "patronym": "Михайлович",
        "department": 14,
        "fullname": "Михайлов Михаил Михайлович",
        "id": 109,
        "position": "специалист",
        "phone": "340-34-34"
    }
];