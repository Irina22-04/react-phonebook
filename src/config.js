export const BASE_URL = 'http://localhost:5000';

//export const BASE_URL = 'https://hidden-badlands-40303.herokuapp.com';