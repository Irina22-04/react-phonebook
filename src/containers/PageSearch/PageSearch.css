.page-container {
    position: relative;
    padding-bottom: 2vh;
    min-height: calc(100vh - 106px);
}

.searchContainer {
    display: flex;
}

.search-buttons {
    display: flex;
    margin-left: 1vw;
    width: 15%;
    min-width: 100px;
}

.search-buttons .button {
    width: 5vw;
    margin-bottom: 2px;
    background-color: #00b5ad;
    min-width: 40px;
}

.search-buttons .active.button {
    width: 6vw;
    background-color: olive;
}

.alphabet-column {
    width: 6.5vw;
    min-width: 50px;
}

.button.all-users-button {
    margin-bottom: 10px;
    min-width: 6vw;

    background-color: #00b5ad;
    margin-left: 1vw;
}

.button.all-users-button:hover,
.search-buttons .button:hover,
.search-buttons .active.button:hover {
    background-color: olive;
}

.ui.pagination.menu {
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    height: 20px;
    top: 100%;
    min-height: 2.4em;
    margin-bottom: 10px;
}

.request-error {
    color: red;
    margin-left: 1vw;
    font-size: 18px;
}

.main-header {
    font-size: 27px;
    width: 100%;
    margin-bottom: 30px;
    text-align: center;
}

@media (max-width: 1000px) {
    .request-error {
        padding-top: 70px;
    }
}

@media (max-width: 480px) {
    .main-header {
        font-size: 22px;
    }
}