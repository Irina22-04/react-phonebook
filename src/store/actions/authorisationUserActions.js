import {get} from "lodash";
import jwt_decode from "jwt-decode";

import {getContactById, loginUser} from "../../utils/api";

export const USER_AUTHORISATION = 'USER_AUTHORISATION';
export const LOGIN_TRY = 'LOGIN_TRY';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAIL = 'LOGIN_FAIL';
export const LEAVE_ACCOUNT = 'LEAVE_ACCOUNT';

export function setUserAuthorisation(dispatch, authorisationState) {
    return dispatch({
        type: USER_AUTHORISATION,
        payload: authorisationState,
    })
}

export const loginUserTry = (dispatch, user) => async (dispatch) => {
    dispatch({type: LOGIN_TRY});
    return await loginUser(user)
        .then(({data}) => {
            const {sub} = jwt_decode(data.accessToken);
            return sub
        })
        .then(async (id) => await getContactById(id))
        .then(({data}) => {
            dispatch({
                type: LOGIN_SUCCESS,
                payload: data,
            })
        })
        .catch(e => {
            dispatch({
                type: LOGIN_FAIL,
                payload: get(e, ['response', 'data', 'error_message'], e.message)
            });
            return Promise.reject('Ошибка авторизации')
        });
};

export function leaveAccount(dispatch) {
    return dispatch({
        type: LEAVE_ACCOUNT,
    })
}