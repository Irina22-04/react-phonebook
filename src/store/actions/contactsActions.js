import {get} from "lodash";

import {getAllContacts} from "../../utils/api";

export const CONTACTS_LOADING = 'CONTACTS_LOADING';
export const CONTACTS_SET = 'CONTACTS_SET';
export const CONTACTS_ERROR = 'CONTACTS_ERROR';
export const CONTACTS_SET_PAGE = 'CONTACTS_SET_PAGE';
export const CONTACTS_SET_SURNAME_LETTER = 'CONTACTS_SET_SURNAME_LETTER';
export const CONTACTS_SET_FULLNAME_PATTERN = 'CONTACTS_SET_FULLNAME_PATTERN';
export const RESET_CONTACT_SEARCH = 'RESET_CONTACT_SEARCH';


export function getAllContactsTry(dispatch) {
    return async function () {
        dispatch({type: CONTACTS_LOADING});
        await getAllContacts()
            .then(data => {
                dispatch({
                    type: CONTACTS_SET,
                    payload: data.data,
                })
            })
            .catch(e => {
                dispatch({
                    type: CONTACTS_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
            });
    }
}

export function setPage(dispatch, page) {
    return dispatch({type: CONTACTS_SET_PAGE, payload: page});
}

export function setSurnameLetter(dispatch, letter) {
    return dispatch({type: CONTACTS_SET_SURNAME_LETTER, payload: letter});
}

export function setFullnamePattern(dispatch, pattern) {
    return dispatch({type: CONTACTS_SET_FULLNAME_PATTERN, payload: pattern});
}

export function resetSearchParams(dispatch) {
    return dispatch({type: RESET_CONTACT_SEARCH});
}
