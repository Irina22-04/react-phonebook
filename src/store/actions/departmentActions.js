import {get} from "lodash";

import {addStructure, deleteStructure, getStructure, updateStructure} from "../../utils/api";

export const STRUCTURE_LOADING = 'STRUCTURE_LOADING';
export const STRUCTURE_SET = 'STRUCTURE_SET';
export const STRUCTURE_ERROR = 'STRUCTURE_ERROR';
export const EDIT_DEPARTMENT = 'SET_EDIT_DEPARTMENT';
export const SET_DEPARTMENT = 'SET_DEPARTMENT';
export const STRUCTURE_UPDATE = 'STRUCTURE_UPDATE';
export const STRUCTURE_DELETE = 'STRUCTURE_DELETE';
export const SET_PARENT_DEPARTMENT = 'SET_PARENT_DEPARTMENT';
export const STRUCTURE_ADD = 'STRUCTURE_ADD';
export const CARRY_DEPARTMENT = 'CARRY_DEPARTMENT';
export const SET_DELETE_DEPARTMENT_ERROR = 'SET_DELETE_DEPARTMENT_ERROR';
export const REMOVE_DELETE_DEPARTMENT_ERROR = 'REMOVE_DELETE_DEPARTMENT_ERROR';

export function setDepartment(dispatch, department) {
    return dispatch({
        type: SET_DEPARTMENT,
        payload: department,
    })

}

export function editDepartment(dispatch, department) {
    return dispatch({
        type: EDIT_DEPARTMENT,
        payload: department,
    })

}

export function getStructureTry(dispatch) {
    return async function () {
        dispatch({type: STRUCTURE_LOADING});
        await getStructure()
            .then(data => {
                dispatch({
                    type: STRUCTURE_SET,
                    payload: data.data,
                })
            })
            .catch(e => {
                dispatch({
                    type: STRUCTURE_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
            });
    }
}

export function updateStructureTry(dispatch, department) {
    return async function () {
        await updateStructure(department)
            .then(data => {
                dispatch({
                    type: STRUCTURE_UPDATE,
                    payload: data.data,
                })
            })
            .catch(e => {
                return Promise.reject(e.message);
            });
    }
}

export function deleteStructureTry(dispatch, department) {
    return async function () {
        await deleteStructure(department)
            .then(() => {
                dispatch({
                    type: STRUCTURE_DELETE,
                    payload: department,
                })
            })
            .catch(e => {
                return Promise.reject(e.message);
            });
    }
}

export function addChildrenDepartment(dispatch, parentDepartment) {
    return dispatch({
        type: SET_PARENT_DEPARTMENT,
        payload: parentDepartment,
    })

}

export function addStructureTry(dispatch, department) {
    return async function () {
        await addStructure(department)
            .then((data) => {
                dispatch({
                    type: STRUCTURE_ADD,
                    payload: data.data,
                });
            })
            .catch(e => {
                return Promise.reject(e.message);
            });
    }
}

export function carryDepartment(dispatch, department) {
    return dispatch({
        type: CARRY_DEPARTMENT,
        payload: department,
    })
}

export function deleteDepartmentError(dispatch, error) {
    return dispatch({
        type: SET_DELETE_DEPARTMENT_ERROR,
        payload: error,
    })
}

export function removeDeleteDepartmentError(dispatch) {
    return dispatch({
        type: REMOVE_DELETE_DEPARTMENT_ERROR,
    })
}
