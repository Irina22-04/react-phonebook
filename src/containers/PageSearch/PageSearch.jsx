import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {Button, Pagination} from "semantic-ui-react";

import Loader from "../../components/Loader/Loader";
import SearchField from "../../components/PageSearch/SearchField/SearchField";
import SearchContainer from "../../components/PageSearch/SearchContainer/SearchContainer";
import {
    getAllContactsTry,
    resetSearchParams,
    setFullnamePattern,
    setPage,
    setSurnameLetter
} from "../../store/actions/contactsActions";
import {getVisibleContacts} from "../../store/selectors/contactSelectors";
import "./PageSearch.css";

class PageSearch extends React.PureComponent {

    static propTypes = {
        contacts: PropTypes.object.isRequired,
    };

    constructor() {
        super();
        this.state = {
            clickedLetter: '',
        }
    }

    componentDidMount() {
        this.getAllClients();
        if (this.props.location.search) {
            const search = new URLSearchParams(this.props.location.search);
            const surnameLetter = search.get("s");
            const fullnamePattern = search.get("f");
            const pageNumber = search.get("page");

            if (surnameLetter) {
                this.props.dispatch(setSurnameLetter(this.props.dispatch, surnameLetter));
                this.setState({clickedLetter: surnameLetter});
            }
            if (fullnamePattern) {
                this.props.dispatch(setFullnamePattern(this.props.dispatch, fullnamePattern));
            }
            if (pageNumber) {
                const page = /^\d+$/.test(pageNumber) ? pageNumber : 1;
                this.props.dispatch(setPage(this.props.dispatch, page));
            }
            return null;
        }
        const {contacts: {paginationOptions: {page}, searchOptions: {surnameLetter, fullNamePattern}}} = this.props;
        const surnameSearchParam = surnameLetter ? `s=${surnameLetter}&` : '';
        const fullnameSearchParam = fullNamePattern ? `f=${fullNamePattern}&` : '';
        this.props.history.push(`?${surnameSearchParam}${fullnameSearchParam}page=${page}`);
    };

    clickLetter = (letter) => {
        this.props.history.push(`?s=${letter}&page=1`);
        this.setState({clickedLetter: letter});
        return this.props.dispatch(setSurnameLetter(this.props.dispatch, letter));
    };

    clickSearch = (searchLetters) => {
        this.props.history.push(`?f=${searchLetters}&page=1`);
        this.setState({clickedLetter: ''});
        return this.props.dispatch(setFullnamePattern(this.props.dispatch, searchLetters));
    };

    getAllClients = () => {
        this.props.dispatch(getAllContactsTry(this.props.dispatch));
    };

    resetSearchParams = () => {
        this.props.dispatch(resetSearchParams(this.props.dispatch));
        this.setState({clickedLetter: ''});
        this.getAllClients();
        this.props.history.push('?page=1');
    };

    handlePageChange = (e, {activePage}) => {
        const pattern = /^\?(s|f)=.*&/gi;
        const firstPart = pattern.test(decodeURI(this.props.location.search));
        const searchParam = (firstPart ? firstPart : '?') + `page=${activePage}`;
        this.props.history.push(searchParam);
        return this.props.dispatch(setPage(this.props.dispatch, activePage));
    };

    getTotalPages = () => {
        const {contacts: {totalElements, paginationOptions: {limit}}} = this.props;
        return Math.ceil(totalElements / limit);
    };

    showSearchForm = () => {
        return this.props.contacts.status <= 1 ? <Loader/> : <SearchField searchFunc={this.clickSearch}/>
    };

    createSearchContainer = () => {
        return this.props.contacts.ctatus <= 2 ? null :
            <SearchContainer cbClickLetter={this.clickLetter} clickedLetter={this.state.clickedLetter}/>
    };

    render() {

        const {contacts: {paginationOptions: {page}}} = this.props;
        return this.props.contacts.status === 2 ? <div className={'request-error'}>{this.props.contacts.error}</div> : (
            <div className={'page-container'}>
                <div className={'main-header'}>Телефонный справочник <br/>ООО "Дети дикой обезьяны"</div>
                {this.showSearchForm()}
                <Button className={'all-users-button'} size='small' onClick={this.resetSearchParams}>Все</Button>
                {this.createSearchContainer()}

                <Pagination
                    activePage={page}
                    onPageChange={this.handlePageChange}
                    totalPages={this.getTotalPages()}
                />
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        contacts: getVisibleContacts(state),
    };
}

export default connect(mapStateToProps)(PageSearch);