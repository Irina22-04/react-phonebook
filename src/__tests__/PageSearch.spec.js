import React from "react";
import renderer, {act} from "react-test-renderer";
import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";

import PageSearch from "../containers/PageSearch/PageSearch";
import {CONTACTS_LOADING, CONTACTS_SET, getAllContactsTry} from "../store/actions/contactsActions";
import {mockUsers} from "../__mocks__/mockUsers";
import {BASE_URL} from "../config";

const initialState = {
    status: 0,
    contacts: [],
    error: '',
    paginationOptions: {
        page: 1,
        limit: 15
    },
    searchOptions: {
        surnameLetter: '',
        fullNamePattern: ''
    }
};


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('PageSearch', () => {
    const state = {contactsForSearch: initialState};
    const store = mockStore(() => state);
    const component = renderer.create(<Provider store={store}><PageSearch location={{search: null}}
                                                                          history={[]}/></Provider>);

    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    it('dispatch CONTACTS_LOADING after render', () => {
        expect(store.getActions()).toEqual([{type: CONTACTS_LOADING}]);
    });

    it('getAllContactsTry dispatch CONTACTS_SET', () => {
        const mock = new MockAdapter(axios);
        const url = `${BASE_URL}/users?_sort=surname`;

        mock.onGet(url).reply(200, mockUsers);

        const expectedActions = [
            {type: CONTACTS_LOADING},
            {type: CONTACTS_LOADING},
            {type: CONTACTS_SET, payload: mockUsers}
        ];

        store.dispatch(getAllContactsTry(store.dispatch)).then(() => {
            expect(expect(store.getActions()).toEqual(expectedActions))
        });

    });

    it('button letter should work', () => {

        const buttons = component.root.findAllByProps({className: 'ui mini button'});
        buttons[4].props.onClick({target: {textContent: "A"}});

        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });

    it('button all should work', () => {

        const button = component.root.findByProps({className: 'all-users-button'});
        button.props.onClick();

        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });

});



