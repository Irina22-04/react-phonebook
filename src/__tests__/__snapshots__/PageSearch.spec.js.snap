// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="page-container"
>
  <div
    className="main-header"
  >
    Телефонный справочник 
    <br />
    ООО "Дети дикой обезьяны"
  </div>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui inverted loader"
      />
    </div>
  </div>
  <button
    className="ui small button all-users-button"
    onClick={[Function]}
  >
    Все
  </button>
  <div
    className="searchContainer"
  >
    <div
      className="search-buttons"
    >
      <div
        className="alphabet-column"
      >
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            А
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Б
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            В
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Г
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Д
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Е
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ж
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            З
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            И
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            К
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Л
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            М
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Н
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            О
          </button>
        </div>
      </div>
      <div
        className="alphabet-column"
      >
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            П
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Р
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            С
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Т
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            У
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ф
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Х
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ц
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ч
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ш
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Щ
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Э
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ю
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Я
          </button>
        </div>
      </div>
    </div>
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui inverted loader"
        />
      </div>
    </div>
  </div>
  <div
    aria-label="Pagination Navigation"
    className="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={1}
    >
      ⟨
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={0}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={0}
    >
      »
    </a>
  </div>
</div>
`;

exports[`PageSearch button all should work 1`] = `
<div
  className="page-container"
>
  <div
    className="main-header"
  >
    Телефонный справочник 
    <br />
    ООО "Дети дикой обезьяны"
  </div>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui inverted loader"
      />
    </div>
  </div>
  <button
    className="ui small button all-users-button"
    onClick={[Function]}
  >
    Все
  </button>
  <div
    className="searchContainer"
  >
    <div
      className="search-buttons"
    >
      <div
        className="alphabet-column"
      >
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            А
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Б
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            В
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Г
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Д
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Е
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ж
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            З
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            И
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            К
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Л
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            М
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Н
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            О
          </button>
        </div>
      </div>
      <div
        className="alphabet-column"
      >
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            П
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Р
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            С
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Т
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            У
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ф
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Х
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ц
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ч
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ш
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Щ
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Э
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ю
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Я
          </button>
        </div>
      </div>
    </div>
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui inverted loader"
        />
      </div>
    </div>
  </div>
  <div
    aria-label="Pagination Navigation"
    className="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={1}
    >
      ⟨
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={0}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={0}
    >
      »
    </a>
  </div>
</div>
`;

exports[`PageSearch button letter should work 1`] = `
<div
  className="page-container"
>
  <div
    className="main-header"
  >
    Телефонный справочник 
    <br />
    ООО "Дети дикой обезьяны"
  </div>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui inverted loader"
      />
    </div>
  </div>
  <button
    className="ui small button all-users-button"
    onClick={[Function]}
  >
    Все
  </button>
  <div
    className="searchContainer"
  >
    <div
      className="search-buttons"
    >
      <div
        className="alphabet-column"
      >
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            А
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Б
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            В
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Г
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Д
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Е
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ж
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            З
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            И
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            К
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Л
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            М
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Н
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            О
          </button>
        </div>
      </div>
      <div
        className="alphabet-column"
      >
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            П
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Р
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            С
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Т
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            У
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ф
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Х
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ц
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ч
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ш
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Щ
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Э
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Ю
          </button>
        </div>
        <div>
          <button
            className="ui mini button"
            onClick={[Function]}
          >
            Я
          </button>
        </div>
      </div>
    </div>
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui inverted loader"
        />
      </div>
    </div>
  </div>
  <div
    aria-label="Pagination Navigation"
    className="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={1}
    >
      ⟨
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={0}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={0}
    >
      »
    </a>
  </div>
</div>
`;
